### PR TITLE
fix(docs): overall improvements

### PR DESCRIPTION
**Pull request by jackmu95**
### Describe the PR

This PR makes some overall improvements to the docs.

### PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [x] Enhancement
- [ ] ARIA accessibility
- [x] Documentation update
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples, fix typos`, `chore: fix typo in README`, etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] New/updated tests are included and passing (if required)
- [ ] Existing test suites are passing
- [ ] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: docs/markdown/reference/router-links/README.md [hunk 1]
```diff
@@ -142,7 +142,7 @@ value can also be configured globally via the `linkExactActiveClass` router cons
 
 When BootstrapVue detects that your app is running under [Nuxt.js](https://nuxtjs.org), it will
 render a [`<nuxt-link>`](https://nuxtjs.org/api/components-nuxt-link) sub component instead of a
-`<router-link>`. `<nuxt-link>` supports all of the above router link props, plus the following
+`router-link>`. `<nuxt-link>` supports all of the above router link props, plus the following
```
Comment (tmorehouse) [inline docs/markdown/reference/router-links/README.md]: Missing leading `<` on router-link

**Comment by tmorehouse** [comment]
Looks like the Right TOC is not rendering content as HTML: 

image

It is escaping the `<` and `>` symbols

**Comment by tmorehouse** [comment]
And the TOC sidebar is no longer collapsing sections (they are all expanded):

image

Which might be due to the custom styles no longer being applied to it

**Comment by tmorehouse** [comment]
Just noticed that scrollspy is not detecting the newer nested structure of the TOC section, keeping the H2 links active when the H3 links are also active.

Will check on what has changed in the structure.

Typically is looks for the following sibling structure:
```html
  ul
    <li>...</li> <!-- h2 -->
    <li>...</li> <!-- h2 -->
    <li>...</li> <!-- h2 -->
    <ul> <!-- subsection of previous h2 -->
      <li>...</li> <!-- h3 -->
      <li>...</li> <!-- h3 -->
    </ul>
    <li>...</li> <!-- h2 -->
    <ul> <!-- subsection of previous <li> -->
      <li>...</li> <!-- h3 -->
      <li>...</li> <!-- h3 -->
    </ul>
</ul>
```

**Comment by jackmu95** [comment]
@tmorehouse The new structure is the one used by Bootstrap.

**Comment by tmorehouse** [comment]
But it doesn't follow their recommended structure for ScrollSpy navs

**Comment by tmorehouse** [comment]
The LI with the nested UL is not receiving the 'active' class (although the `nav-link` inside the LI is)

**Comment by tmorehouse** [comment]
Just checked ScrollSpy code at Bootstrap V4 repo, and they have the same issue with scrollspy not setting the active state on the nested `UL`'s parent` LI.nav-item` element (although the `a.nav-link` does get the active state class)

**Pull request merged**